File: lib/data/migration/ids.dart
import 'package:uuid/uuid.dart';

/// Namespace fijo de Cingula. Generado una sola vez; NUNCA cambiar: los uuid v5
/// derivados de el son la identidad de las obras y portales que crean, por
/// separado y sin hablarse, la migracion del celular y el script de Neon (D-24).
const String cingulaNamespace = '6f1a6c3e-9b54-4b2a-8f3d-1d0c9a7e5b21';

String obraUuidForPath(String pathUuid) =>
    const Uuid().v5(cingulaNamespace, 'obra:$pathUuid');

String portalUuidForTrigger(String triggerUuid) =>
    const Uuid().v5(cingulaNamespace, 'portal:$triggerUuid');

File: backend/api/_lib/ids.js
import { createHash } from 'node:crypto';

// Namespace fijo de Cingula: NUNCA cambiar (identidad de obras/portales, D-24).
export const CINGULA_NAMESPACE = '6f1a6c3e-9b54-4b2a-8f3d-1d0c9a7e5b21';

export function uuidV5(namespace, name) {
  const ns = Buffer.from(namespace.replace(/-/g, ''), 'hex');
  const bytes = createHash('sha1')
    .update(Buffer.concat([ns, Buffer.from(name, 'utf8')]))
    .digest()
    .subarray(0, 16);
  bytes[6] = (bytes[6] & 0x0f) | 0x50;
  bytes[8] = (bytes[8] & 0x3f) | 0x80;
  const h = bytes.toString('hex');
  return `${h.slice(0, 8)}-${h.slice(8, 12)}-${h.slice(12, 16)}-${h.slice(16, 20)}-${h.slice(20)}`;
}

export const obraUuidForPath = (pathUuid) => uuidV5(CINGULA_NAMESPACE, 'obra:' + pathUuid);
export const portalUuidForTrigger = (triggerUuid) => uuidV5(CINGULA_NAMESPACE, 'portal:' + triggerUuid);
